lib/codegen/src/licm.rs

```diff
@@ -90,11 +90,18 @@ fn create_pre_header(
         }
     }
     {
-        let mut pos = EncCursor::new(func, isa).at_top(header);
+        let mut pos = FuncCursor::new(func).at_top(header);
         // Inserts the pre-header at the right place in the layout.
         pos.insert_ebb(pre_header);
         pos.next_inst();
-        pos.ins().jump(header, pre_header_args_value.as_slice(pool));
+        let jump = pos.ins().jump(header, pre_header_args_value.as_slice(pool));
+        if !pos.func.encodings.is_empty() {
```

Comment:
Why wasn't the EncCursor enough here? (In any case, this hunk could be moved to the previous commit)